crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -190,6 +241,16 @@ impl SourceMap {
         Ok(rope.cell())
     }
 
+    #[turbo_tasks::function]
+    pub async fn tokens(self: Vc<Self>) -> Result<Vc<Tokens>> {
+        let crate_map = match &*self.await? {
+            Self::Regular(m) => m.deref().deref().deref().to_owned(),
+            Self::Sectioned(m) => m.flatten().await?.deref().deref().deref().to_owned(),
+        };
+
+        Ok(Vc::cell(crate_map.tokens().map(|t| t.into()).collect()))
```

Comment:
This is 2 clones of the tokens (one of the full map, then another as we convert into `Token`. Do we need to clone anything here? Can't we just refer to the original's tokens?

Reply:
This is part of the public turbopack SourceMap interface, so I figured it should return those tokens.